corehq/apps/case_migrations/views.py

```diff
@@ -81,6 +91,15 @@ def migration_restore(request, domain, case_id):
             # Formplayer will be creating these cases for the first time, so
             # include create blocks
             content.append(get_case_element(case, ('create', 'update'), V2))
+        if ADD_LIMITED_FIXTURES_TO_SMS_WORKFLOW.enabled(domain):
+            _add_limited_fixtures(domain, case_id, content)
         payload = content.get_fileobj()
 
     return RestoreResponse(payload).get_http_response()
+
+
+def _add_limited_fixtures(domain, case_id, content):
+    serializer = FlatLocationSerializer()
+    data_fields = get_location_data_fields(domain)
+    restore_user = MockCaseAsRestoreUser(user_id=case_id)
```

Comment:
Mocking the case as the user here, to be able for the fixture element to add user_id later, https://github.com/dimagi/commcare-hq/blob/a58647d05233690ad7d1bbced8dd29255dd8e0cc/corehq/apps/locations/fixtures.py#L164 and another occurrence in the same method

Reply:
This mock seems brittle - it'd break if someone attempted to use any other attribute of `restore_user` - doesn't it try to access `domain`?
https://github.com/dimagi/commcare-hq/blob/c3a26972498c0f648ec95dd8acf78c5c6f426383/corehq/apps/locations/fixtures.py#L150-L152

Why don't you instead modify the serializers to accept `domain` and `user_id` args directly, making it clearer what the real dependencies are?

Reply:
RestoreUser is something that exists in context of a restore. And the case here is like the user, hence the thought of mocking the case as the user. Restore code does pass the restore user around and store the relevant props on it, and passing domain and user_id separately does conflict with that practice. No?
So doesn't it make sense for the fixture generator to be provided with a restore user like object.

I missed the domain bit. My attempt would be to add the domain to the mocked object. And i think any other thing that is, added in the restore user and is needed for fixture generation, should be added to the mock. It will be brittle in that sense always since any changes to restore code would only make changes to RestoreUser and not worry about this custom logic, specific to this place where we generate the restore content outside the HQ default way of generating restore.

Reply:
True, but this isn't a real restore, and there isn't a real user.  This code doesn't actually use the fixture_generator either, just this `FlatLocationSerializer`, which is responsible for generating an xml fixture, but doesn't it doesn't need to operate on a `restore_state` or maintain compatibility with anything else outside of `locations/fixtures.py`, so I'd just go with the simplest approach.

The same arguments go for passing around `request` - it does contain the parameters of a request, but I try to minimize how much it's passed around, as it obscures what attributes are actually needed.  And it's definitely preferable to refactor shared code to not rely on a request object vs assembling a mock request that doesn't go through middleware and is missing a bunch of stuff.
*cough* [reports](https://github.com/dimagi/commcare-hq/blob/c3a26972498c0f648ec95dd8acf78c5c6f426383/corehq/apps/reports/generic.py#L199-L222) *cough*

Reply:
makes sense, thank you :)

Reply:
https://github.com/dimagi/commcare-hq/pull/29482/commits/859bc8d55c04c1f8b04a04cb614e7a4660def159

and 

https://github.com/dimagi/commcare-hq/pull/29482/commits/4ecb78b02d56db6b8f30e0462f6cfe0e58e8f018